src/viam/module/module.py

```diff
@@ -176,3 +178,14 @@ def add_model_from_registry(self, subtype: Subtype, model: Model):
             Registry.lookup_resource_creator(subtype, model)
         except ResourceNotFoundError:
             raise ValueError(f"Cannot add model because it has not been registered. Subtype: {subtype}. Model: {model}")
+
+    async def validate_config(self, request: ValidateConfigRequest) -> ValidateConfigResponse:
+        config: ComponentConfig = request.config
+        subtype = Subtype.from_string(config.api)
+        model = Model.from_string(config.model)
+        validator = Registry.lookup_validator(subtype, model)
+        if validator is not None:
+            dependencies = validator(config)
```

Comment:
I think we'll want some `try/except` logic around `validator` to catch potential errors from `validator(config)`. We'll want to return a gRPC error instead of raising an error within the Python module.